File: Examples/Java/SDK/src/main/java/com/aspose/imaging/cloud/examples/tiffframes/CropFrame.java
package com.aspose.imaging.cloud.examples.tiffframes;

import com.aspose.imaging.cloud.examples.Utils;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.StandardCopyOption;

public class CropFrame {

    public static void main(String... args) throws IOException {
        String input = "Sample1.tiff";
        Path inputFile = Utils.getPath(CropFrame.class, input);
        String output = "Sample2.tiff";
        Path outputFile = Utils.getPath(CropFrame.class, output);

        Integer frameId = 0;
        Integer newWidth = 200;
        Integer newHeight = 200;
        Integer x = 20;
        Integer y = 20;
        Integer rectWidth = 100;
        Integer rectHeight = 100;
        String rotateFlipMethod = "";
        Boolean saveOtherFrames = false;
        String outPath = "";

        Utils.getStorageSdk().PutCreate(
                input,
                null,
                Utils.STORAGE,
                inputFile.toFile()
        );

        com.aspose.imaging.model.ResponseMessage apiResponse
                = Utils.getImagingSdk().GetImageFrame(
                        input,
                        frameId,
                        newWidth,
                        newHeight,
                        x,
                        y,
                        rectWidth,
                        rectHeight,
                        rotateFlipMethod,
                        saveOtherFrames,
                        outPath,
                        Utils.FOLDER,
                        Utils.STORAGE
                );

        Files.copy(apiResponse.getInputStream(), outputFile, StandardCopyOption.REPLACE_EXISTING);

    }
}

File: Examples/Node.js/SDK/tiffframes/CropFrame.js
var fs = require('fs'),
    utils = require('../utils.js'),
    image = utils.imagingApi;
    storage = utils.storageApi;

var inputName = "sample1.tiff";
var inputPath = utils.getPath(__filename, inputName);

var outputName = "nodejsSDK.output.tiff";
var outputPath = utils.getPath(__filename, outputName);

var frameId = 0;
var newWidth = 200;
var newHeight = 200;
var x = 20;
var y = 20;
var rectWidth = 100;
var rectHeight = 100;
var Storage = null;
var folder = null;

storage.PutCreate(inputName, null, Storage,inputPath, function(responseMessage) {
    image.GetImageFrame(inputName, frameId, newWidth, newHeight, x, y, rectWidth, rectHeight, null, null, null, folder, Storage, function(responseMessage) {
        var writeStream = fs.createWriteStream(outputPath);
        writeStream.write(responseMessage.body);
    });
});
